src/workflow/CommandExecutor.py

```
@@ -73,12 +73,15 @@ def run_command(self, command: list[str]) -> None:
         # Ensure all command parts are strings
         command = [str(c) for c in command]
 
+        # Log cwd
+        if cwd is not None:
+            self.logger.log(f'Preparing command... cwd is set to \'{cwd}\'')
         # Log the execution start
         self.logger.log(f"Running command:\n"+' '.join(command)+"\nWaiting for command to finish...", 1)   
```

Flake8 (7.2.0)

[error] f-string is missing placeholders

(F541)